web/__init__.py

```diff
@@ -1,23 +1,43 @@
+import os
+import re
 import json
+import tempfile
 import logging as log
 
-from flask import Flask, request, render_template, Response
+from secrets import token_urlsafe as token
+
+from flask import Flask, flash, request, redirect, url_for, render_template, Response, session
 from flask_sock import Sock
+from werkzeug.utils import secure_filename
+
+import libflagship.httpapi
+import libflagship.logincache
 
 from libflagship.pppp import P2PSubCmdType, FileTransfer
 from libflagship.ppppapi import FileUploadInfo, PPPPError
 
 
 from web.lib.service import ServiceManager
 
+from user_agents import parse
+
 import cli.util
+import cli.config
+
+
+def dict_match(dict, match):
+    for key in dict:
+        if re.match(rf"^{key}.*", match):
+            return key
+
 
 app = Flask(
     __name__,
     root_path=".",
     static_folder="static",
     template_folder="static"
 )
+app.secret_key = token(24)
```

Comment:
What's this used for? Isn't this for cookie encryption (at least among other things?)

Reply:
That makes sense - maybe add

```python3
# required by the flash() function
app.secret...
```